### ISSUE TITLE
JTextArea doesn't update in the Main GUI

### ISSUE THREAD

**Issue by Rovinovic** [Rovinovic/SEC#2]
JTextArea opens three GUI's instead of updating in the main GUI.

Need to update the main GUI JTextArea only.

**Comment by afischer12** [comment]
Will create a fix for this.

**Comment by afischer12** [comment]
I'm sorry, can you point me to the problematic file? I also need the variable name of the jtextarea that's causing issues and if possible the function that is supposed to update it but is opening 3 new guis instead.

**Comment by Rovinovic** [comment]
@afischer12 Thanks for helping mate. 

The Three GUIS are opening from the three packages FOX, NYT and WASH.
The Main File running is the MainFrame.java in the Main package.

MainFrame calls the 3 classes of the three packages.

What happens is the extraction part of the three packages happens in three GUI's

What i need is this to happen in the MainFrame.java's Main GUI.

**Comment by Rovinovic** [comment]
@afischer12 Run the project and and wait for few seconds to download the web page and extract the file. It will create three GUI's

**Comment by afischer12** [comment]
The problem is because each of FOX NYT and WASH use the line "new MainFrame()", which is the cause of it opening the 3 GUIs instead of just one. When the main() method of MainFrame is called, it creates a new MainFrame object that creates new instances of the 3 classes via their constructors when the  java.awt.EventQueue.invokeAndWait method is used. These three all create a new mainframe object in a variable and set them visible, which is the cause of the 3 guis. What I don't understand is what exactly is supposed to be done, since the only comments are commented out lines of code. Can you explain exactly what each class is supposed to do?

**Comment by Rovinovic** [comment]
@afischer12 yes. thats the issue here and I can't seem to find a solution for this. 

The NYT FOX and WASH Packages are identical and they just extract the homepage of the three websites and display the headlines.

In order to access the mainframe Swing component jtextarea1, I created the MainFrame class in all three. I Cannot find any other solution to access the jtextarea swing component to another class. 

And in the MainFrame main class inside the **java.awt.EventQueue.invokeAndWait(new Runnable() {** happens the code of displaying the content.

If I can access the swing component jtextarea with a different option than making the MainFrame class in the three packages, that would be the ideal one.

In the three packages, under jsoup there's the jtextarea append happens.

If there's any method which I can access the jtextarea in a different way and update the MainFrame GUI only, that would be better.

I hope you got what I said. :)

**Comment by afischer12** [comment]
OK i'll make a commit giving you a possible solution in the editing. You'll have to remove the code that creates the three GUIs and delete it because I don't want to accidentally delete anything important. I won't edit any code in the three classes but I will add a method to the MainFrame class and a comment at the top of each of the 3 classes with the line of code that can access the JTextArea

**Comment by afischer12** [comment]
Hold on what i thought would work actually won't because of how the classes are non-static. I have another solution give me a minute

**Comment by Rovinovic** [comment]
@afischer12 do any changes and commit here mate. No problem. I have a backup with me :)

**Comment by afischer12** [comment]
Just make sure to test my changes and let me know if they work.

**Comment by afischer12** [comment]
When are the DLL methods of each of the three classes called?

**Comment by afischer12** [comment]
nvm found it

**Comment by Rovinovic** [comment]
@afischer12 inside the main method of MainFrame under 

**SwingWorker worker = new SwingWorker() {**

there i've created 3 runnables. 
inside those three, three DLL methods are called.

**Comment by afischer12** [comment]
see if the 4 pull requests i just added work

**Comment by afischer12** [comment]
did they work?

**Comment by Rovinovic** [comment]
@afischer12 I dont see any output coming to the mainframe GUI :/

**Comment by afischer12** [comment]
oh

**Comment by afischer12** [comment]
hold on a sec

**Comment by afischer12** [comment]
try now

**Comment by Rovinovic** [comment]
let me try

**Comment by afischer12** [comment]
I removed the variable that you have to pass, deleted the MainFrame variable in the three classes, changed the three DLL methods' output type to String, return the String you want to append, and had the method that calls the three DLL methods do the jTextArea1.append

**Comment by Rovinovic** [comment]
Still, it doesn't pass it to the main GUI :/ you could try it, just run the project from netbeans or the IDE you're using.

**Comment by afischer12** [comment]
ok one sec. I'll find the problem

**Comment by afischer12** [comment]
can you provide me with the jsoup library?

**Comment by Rovinovic** [comment]
@afischer12 this is the library, add the jar file to the project as an external jar 

https://jsoup.org/download

**Comment by afischer12** [comment]
thx

**Comment by afischer12** [comment]
alright well I will continue working on this tomorrow. Unfortunately the fact that the calls to DLL are done in an object other than the MainFrame is confusing the issue and I have to find a workaround for that.

**Comment by Rovinovic** [comment]
@afischer12 did some tweaks, still its not coming to the main GUI. Anyways thanks a lot for the effort boss. Really appreciate it.  :)  I'll try to fix it from my end as well. if you find any solution in the meantime, feel free to commit. Thanks a lot :)

**Comment by afischer12** [comment]
ok i got it so the strings from the dll calls are now stored outside the other object, but now I'm facing a new problem: how to get a variable that is created in a static reference non-static. The entire code may have to be modified to allow the dll methods to be non-static

**Comment by Rovinovic** [comment]
@afischer12 Hi mate, thanks a lot for your efforts. I found the solution for your one. instead of a string, an array was returned and appended to the main GUI. now its only shows in the main GUI. Many many thanks to you for the help. If you can, please delete the folk of this repo. ill be deleting this repo as well since this was a small group project. Thanks million times :)

**Comment by afischer12** [comment]
np :) Glad i could help

**Comment by afischer12** [comment]
deleted my fork
